### PR TITLE
Fix client filepaths

### PR DESCRIPTION
**Pull request by pngwn**
# Description

File paths returned by the client were absolute paths for the server they were running on. However, due to how we handle filepaths, reconstructing them is tricky. This PR handles that automatically for client users (in most cases at least).

I've currently disabled this via a flag for the core gradio app for now, it would be good to shift all of this work to the client but that will require some internal refactoring, and i think it would be simpler to wait until we clean up the signatures in 4.0.

Please include: 
* relevant motivation
* a summary of the change 
* which issue is fixed. 
* any additional dependencies that are required for this change.

Closes: # (issue)

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added a short summary of my change to the CHANGELOG.md
- [x] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# A note about the CHANGELOG

Hello 👋 and thank you for contributing to Gradio!

All pull requests must update the change log located in CHANGELOG.md, unless the pull request is labeled with the "no-changelog-update" label.

Please add a brief summary of the change to the Upcoming Release > Full Changelog section of the CHANGELOG.md file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

If you would like to elaborate on your change further, feel free to include a longer explanation in the other sections.
If you would like an image/gif/video showcasing your feature, it may be best to edit the CHANGELOG file using the 
GitHub web UI since that lets you upload files directly via drag-and-drop.

### CODE REVIEW

**Comment by pngwn** [comment]
I think this should work:

```
cd js/client
pnpm i 
pnpm build 
```

Create a test.js file like this:

```js
import { client } from "./dist/index.js";

async function run() {
	const app = await client("gradio/fake_gan");

	console.log(info);
	const result = await app.predict(0, []));

	// @ts-ignore
	console.log(result?.data);
}

run();
```

```
node test.js
```

The output should be full urls (starting with `https://` rather than `/`).

**Review by aliabid94 — COMMENTED** [review]
What I don't like about this approach is that for every single component that returns file paths differently, we have to add logic to `transform_outputs`. It also assumes that any object that has the `is_file` property is a File object, which could be dangerous. 
I want to suggest a different approach. What if when you make an api call to a gradio backend, the backend can also receive a "root_url", and if it receives this, the postprocess functions of the component handle prepending the root url to the filepaths they return. This will keep the file path logic coupled with the component itself. 
This would not be backwards compatible with existing gradio spaces, so we could have the logic in this PR as a backup if the gradio space is older.

File: .changeset/nice-tools-wait.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
```
[the remark this answers is not in the document]
Comment (pngwn) [reply .changeset/nice-tools-wait.md:1]: This is how we manage versions and the change log for anything we are publishing to npm. It needs to be in the root.

**Comment by pngwn** [comment]
Yeah I don't like it much, either. I'd actually prefer if we always just returned full URLs from the server, the server should have all of the necessary information to do that.

**Comment by pngwn** [comment]
Would your approach work for these  'proxy' urls as well as the normal one?

**Comment by aliabid94** [comment]
> Yeah I don't like it much, either. I'd actually prefer if we always just returned full URLs from the server, the server should have all of the necessary information to do that.
from the request headers the backend can infer its own server url, yeah? Can we implement this instead then?

**Comment by pngwn** [comment]
> from the request headers the backend can infer its own server url, yeah? Can we implement this instead then?

Definitely, I'd prefer that approach. Although we will need to update the gradio frontend too, I guess. 

Will close this PR.

**Comment by pngwn** [comment]
Actually, we probably still need this for older gradio version

**Pull request merged**